### DISCUSSION TITLE
Datasource in Custom Federation: Migrating from Wildfly to Quarkus

### DISCUSSION THREAD

**Discussion by simin-c** [keycloak/keycloak#11200] [Q&A]
Hi all,

In the _wildfly_ distribution of keycloak, we were sharing a datasource defined in standalone.xml for both keycloak and a **User federation**

We were accessing the datasource in our code via its jndi-name. However as I understand, jndi is no supported in Quarkus by design.  

How can we still use a data source that is defined in keycloak.conf or quarkus.properties to leverage the datasource and pooling of keycloak.x?

**Comment by eggyknap** [comment]
I have the same problem, except I haven't yet found a file called quarkus.properties :)

**Comment by simin-c** [comment]
That's right. If it does not exist, then you can manually create one Am 14.07.2022 22:04 schrieb Joshua Tolley ***@***.***>:
I have the same problem, except I haven't yet found a file called quarkus.properties :)

**Comment by eggyknap** [comment]
I successfully used hints from https://github.com/keycloak/keycloak/discussions/11464, and added in that thread some details on exactly what I did. Perhaps that will help you.
